tests/test_location.py

```diff
@@ -0,0 +1,192 @@
+    assert set(locations) == {"local"}
+
+    other = tmp_path / "other"
+    with pytest.raises(Exception) as e:
+        orderly_location_add("other", "path", {"path": str(other)}, root=root)
+
+    assert e.match("Expected 'path' to be an existing directory")
+
+    other.mkdir()
+    with pytest.raises(Exception) as e:
+        orderly_location_add("other", "path", {"path": str(other)}, root=root)
+
+    assert e.match("Did not find existing outpack root in .*")
+
+
+def test_can_rename_a_location(tmp_path):
+    root = {}
+    for name in ["a", "b", "c"]:
+        root[name] = create_temporary_root(tmp_path / name)
+
+    orderly_location_add(
+        "b", "path", {"path": str(root["b"].path)}, root=root["a"]
+    )
+
+    locations = orderly_location_list(root=root["a"])
+    assert set(locations) == {"local", "b"}
+
+    orderly_location_rename("b", "c", root=root["a"])
+
+    locations = orderly_location_list(root=root["a"])
+    assert set(locations) == {"local", "c"}
+
+
+def test_cant_rename_a_location_using_existent_name(tmp_path):
+    root = {}
+    for name in ["a", "b", "c"]:
+        root[name] = create_temporary_root(tmp_path / name)
+
+    orderly_location_add(
+        "b", "path", {"path": str(root["b"].path)}, root=root["a"]
+    )
+    orderly_location_add(
+        "c", "path", {"path": str(root["c"].path)}, root=root["a"]
+    )
+
+    with pytest.raises(Exception) as e:
+        orderly_location_rename("b", "c", root["a"])
+
+    assert e.match("A location with name 'c' already exists")
+
+    with pytest.raises(Exception) as e:
+        orderly_location_rename("b", "local", root["a"])
+
+    assert e.match("A location with name 'local' already exists")
+
+
+def test_cant_rename_a_non_existent_location(tmp_path):
+    root = create_temporary_root(tmp_path)
+    assert set(orderly_location_list(root=root)) == {"local"}
+
+    with pytest.raises(Exception) as e:
+        orderly_location_rename("a", "b", root)
+
+    assert e.match("No location with name 'a' exists")
+
+
+def test_cant_rename_default_locations(tmp_path):
+    root = create_temporary_root(tmp_path)
+
+    with pytest.raises(Exception) as e:
+        orderly_location_rename("local", "desktop", root)
+
+    assert e.match("Cannot rename default location 'local'")
+
+
+def test_can_remove_a_location(tmp_path):
+    root = {}
+    for name in ["a", "b", "c"]:
+        root[name] = create_temporary_root(tmp_path / name)
+
+    orderly_location_add(
+        "b", "path", {"path": str(root["b"].path)}, root=root["a"]
+    )
+    orderly_location_add(
+        "c", "path", {"path": str(root["c"].path)}, root=root["a"]
+    )
+
+    locations = orderly_location_list(root=root["a"])
+    assert set(locations) == {"local", "b", "c"}
+
+    # Remove a location without packets
+    orderly_location_remove("c", root=root["a"])
+
+    locations = orderly_location_list(root=root["a"])
+    assert set(locations) == {"local", "b"}
+
+    ## TODO:
```

Comment:
can you make tickets for all of these (check all the similar comments in the PR - I think I generated 10 in my last PR!)

Reply:
Done that too https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4601/Orphan-packets-when-a-location-is-removed